server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -134,9 +140,20 @@ public synchronized void onNewCheckpoint(final ReplicationCheckpoint receivedChe
             );
             return;
         }
+        boolean checkpointFromNewPrimary = false;
+
+        if (receivedCheckpoint.getPrimaryTerm() > pTermLinkedToCopiedFiles) {
+            logger.info(
+                "Incoming checkpoint primary term {} higher compared to existing primary term {}",
+                receivedCheckpoint.getPrimaryTerm(),
+                pTermLinkedToCopiedFiles
+            );
+            pTermLinkedToCopiedFiles = receivedCheckpoint.getPrimaryTerm();
+            checkpointFromNewPrimary = true;
```

Comment:
How does this behave with the first replication cycle?  It looks to me like this will always allow diff.different files on that first event bc the default value is UNASSIGNED_PRIMARY_TERM? I'd imagine we would want to store a map of shard id to pTerm and init a value here when the first event is received.

Reply:
Thanks @mch2 for pointing this out. Yes, this needs a map to keep track of different shard id's primary terms.